selfdrive/pandad/pandad.py

```diff
@@ -62,12 +63,21 @@ def flash_panda(panda_serial: str) -> Panda:
 
 
 def main() -> NoReturn:
+  # override manager's immediate SystemExit to allow pandad to exit gracefully
+  def signal_handler(signum, frame):
+    nonlocal do_exit
+    do_exit = True
+    cloudlog.warning(f"caught signal {signum}, exiting")
+
+  do_exit = False
+  signal.signal(signal.SIGTERM, signal_handler)
+
   count = 0
   first_run = True
   params = Params()
   no_internal_panda_count = 0
 
-  while True:
+  while not do_exit:
```

Comment:
This behavior matches pandad.cc's of looping until it gets a SIGTERM signal and then gracefully exiting.